src/module-elasticsuite-catalog/Model/ResourceModel/Product/Indexer/Fulltext/Datasource/InventoryData.php

```diff
@@ -93,7 +93,8 @@ public function loadInventoryData($storeId, $productIds)
                     'stock_status'  => 'stock_index.' . IndexStructure::IS_SALABLE,
                     'qty'           => 'stock_index.' . IndexStructure::QUANTITY,
                 ]
-            );
+            )
+            ->where('product.entity_id IN(?)', $productIds);
```

Comment:
Could you 

- add a space between "IN" and "(?)" (just to nitpick)
- add a `->group('product.entity_id');` clause after the where ?

On a EE version, there could be staged updates present in the `catalog_product_entity` and otherwise, several (identical, mind you) rows could be retrieved for a single product.